src/flownet/network_model/_mitchell.py

```diff
@@ -111,88 +123,132 @@ def mitchell_best_candidate_modified_3d(
+) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """
+
+    Args:
+        num_candidates: Number of candidates to consider for per additional flow node,
+        x_min: minimum x coordinate of (scaled) perforations,
+        x_max: maximum x coordinate of (scaled) perforations,
+        y_min: minimum y coordinate of (scaled) perforations,
+        y_max: maximum y coordinate of (scaled) perforations,
+        z_min: minimum z coordinate of (scaled) perforations,
+        z_max: maximum z coordinate of (scaled) perforations,
+        perforation_hull: Convex hull of delaunay triangulated (scaled) perforations,
+        concave_hull_bounding_boxes Numpy array with x, y, z min/max boundingboxes for each grid block,
+    Returns:
+        x_candidate: Array of length num_candidates containing x coordinates of candidates,
+        y_candidate: Array of length num_candidates containing y coordinates of candidates,
+        z_candidate: Array of length num_candidates containing z coordinates of candidates,
+
+    """
+    in_hull = np.asarray([False] * num_candidates)
+    x_candidate = np.zeros(num_candidates)
+    y_candidate = np.zeros(num_candidates)
+    z_candidate = np.zeros(num_candidates)
+
+    while not all(in_hull):
+        # Generate a set of random candidates that will be the new
+        # flow nodes
+        x_candidate_tmp = x_min + np.random.rand(num_candidates) * (x_max - x_min)
+        y_candidate_tmp = y_min + np.random.rand(num_candidates) * (y_max - y_min)
+        z_candidate_tmp = z_min + np.random.rand(num_candidates) * (z_max - z_min)
+
+        # Update the list of flow node candidates. Only the points that previously
+        # were not inside the convex hull are updated.
+        np.putmask(x_candidate, np.invert(in_hull), x_candidate_tmp)
+        np.putmask(y_candidate, np.invert(in_hull), y_candidate_tmp)
+        np.putmask(z_candidate, np.invert(in_hull), z_candidate_tmp)
+
+        candidates = np.vstack([x_candidate, y_candidate, z_candidate]).T
+
+        if concave_hull_bounding_boxes is not None:
+            # Test whether all points are inside the bounding boxes of the gridcells of the reservoir volume.
+            # This is the always the case when place_nodes_in_volume_reservoir is True.
+            in_hull = check_in_hull(
+                concave_hull_bounding_boxes, candidates, in_hull_known=in_hull
+            )
+        else:
+            if perforation_hull is not None:
+                # Test whether all points are inside the convex hull of the perforations
+                if np.isclose(z_min, z_max):
+                    in_hull = perforation_hull.find_simplex(candidates[:, (0, 1)]) >= 0
+                else:
+                    in_hull = perforation_hull.find_simplex(candidates) >= 0
+            else:
+                raise ValueError(
+                    "perforation_hull should never be None when concave_hull_bounding_boxes is also None"
```

Comment:
```suggestion
                    "The perforation_hull should never be None when concave_hull_bounding_boxes is also None."
```

Can this actually happen at this point?

Reply:
No I am pretty sure it can't, but pylint still complained. 

`src/flownet/network_model/_mitchell.py:233: error: Item "None" of "Optional[Any]" has no attribute "find_simplex"
src/flownet/network_model/_mitchell.py:235: error: Item "None" of "Optional[Any]" has no attribute "find_simplex"
Found 2 errors in 1 file (checked 61 source files)
Error: Process completed with exit code 1.`

Didn't know how else to fix this.

Reply:
Can you do like this? 
![image](https://user-images.githubusercontent.com/46521401/118935045-5eda9080-b94b-11eb-9b96-c260a9e98a8a.png)

Reply:
Yes this works actually. I got a bit confused because concave_hull_bounding_boxes and perforation_hull can both be true and if that is the case you only want the `concave_hull_bounding_boxes is not None` statement to run. But since that one is the first if statement, that should be fine indeed.